examples/nvram/counter.c

```diff
@@ -0,0 +1,200 @@
+/* counter.c
+ *
+ * Copyright (C) 2006-2022 wolfSSL Inc.
+ *
+ * This file is part of wolfTPM.
+ *
+ * wolfTPM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfTPM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+/* Example for incrementing an NV counter.
+ *
+ * NB: This example uses Parameter Encryption to protect
+ *     the Password Authorization of the TPM NVRAM Index
+ *
+ **/
+
+#include <wolftpm/tpm2_wrap.h>
+
+#include <examples/nvram/nvram.h>
+#include <examples/tpm_io.h>
+#include <examples/tpm_test.h>
+#include <examples/tpm_test_keys.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifndef WOLFTPM2_NO_WRAPPER
+
+/******************************************************************************/
+/* --- BEGIN TPM NV Counter Example -- */
+/******************************************************************************/
+static void usage(void)
+{
+    printf("Expected usage:\n");
+    printf("./examples/nvram/counter [-nvindex=0x0] [-aes/-xor]\n");
+    printf("* -nvindex=[handle] (default 0x%x)\n", TPM2_DEMO_NV_COUNTER_INDEX);
+    printf("* -aes/xor: Use Parameter Encryption\n");
+}
+
+int TPM2_NVRAM_Counter_Example(void* userCtx, int argc, char *argv[])
+{
+    int rc;
+    WOLFTPM2_DEV dev;
+    WOLFTPM2_KEY storage;
+    WOLFTPM2_SESSION tpmSession;
+    WOLFTPM2_HANDLE parent;
+    WOLFTPM2_NV nv;
+    TPMS_NV_PUBLIC nvPublic;
+    int paramEncAlg = TPM_ALG_NULL;
+    word32 nvIndex = TPM2_DEMO_NV_COUNTER_INDEX;
+
+    XMEMSET(&tpmSession, 0, sizeof(tpmSession));
+    XMEMSET(&parent, 0, sizeof(parent));
+    XMEMSET(&nv, 0, sizeof(nv));
+    XMEMSET(&storage, 0, sizeof(storage));
+
+    if (argc >= 2) {
+        if (XSTRNCMP(argv[1], "-?", 2) == 0 ||
```

Comment:
XSTRNCMP not necessary. Just use XSTRCMP. argv guaranteed to be null-terminated strings. Same applies below.